symfony/src/ApiResource/Register.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+namespace App\ApiResource;
+
+use ApiPlatform\Core\Annotation\ApiResource;
+use App\Dto\RegisterOutput;
+use Symfony\Component\Serializer\Annotation\Groups;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * This is a custom API Platform ApiResource, with DataPersister and DataProvider.
+ *
+ * @ApiResource(
+ *     collectionOperations={
+ *         "post_register_api"={
+ *              "path"= "/auth/register",
+ *              "method"="POST",
+ *              "openapi_context"={
+ *                  "summary"="Register a new user",
+ *                  "description"="Register a new user"
+ *              }
+ *         }
+ *     },
+ *     itemOperations={},
+ *     shortName="Register",
+ *     denormalizationContext={
+ *          "groups"={"register:api-write"},
+ *          "swagger_definition_name"="WRITE"
+ *     },
+ *     output=RegisterOutput::class
+ * )
+ */
+class Register
+{
+    /**
+     * @var string
+     *
+     * @Groups({"register:api-write"})
+     *
+     * @Assert\NotBlank()
+     * @Assert\Length(
+     *     min = 1,
+     *     max = 50,
+     *     minMessage="Minimim length 1 characters",
```

Comment:
Voor zaken die we ook echt gaan gebruiken (geen examples) best al translations voorzien

=> algemene icapps translations voor validatie nl/en dan kan dit gewoon mee genomen worden